scene/resources/3d/primitive_meshes.cpp

```diff
@@ -3975,3 +3975,1032 @@ void TextMesh::set_uppercase(bool p_uppercase) {
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "extend_edges", PROPERTY_HINT_NONE, "hint_tooltip:Extend edges to cover the curve."), "set_extend_edges", "is_extend_edges");
+	ADD_PROPERTY(PropertyInfo(Variant::VECTOR3, "up_vector", PROPERTY_HINT_NONE, "hint_tooltip:Up vector for the curve."), "set_up_vector", "get_up_vector");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "follow_curve", PROPERTY_HINT_NONE, "hint_tooltip:Follow the curve's tilt instead of up vector."), "set_follow_curve", "is_follow_curve");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "tessellation_mode", PROPERTY_HINT_ENUM, "Adaptive,Baked,Disabled"), "set_tessellation_mode", "get_tessellation_mode");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "tessellation_tolerance", PROPERTY_HINT_RANGE, "0.001,16.0,0.001,or_greater,suffix:m"), "set_tessellation_tolerance", "get_tessellation_tolerance");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "corner_threshold", PROPERTY_HINT_RANGE, "0.0,180.0,0.1,radians_as_degrees"), "set_corner_threshold", "get_corner_threshold");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "smooth_shaded_corners", PROPERTY_HINT_NONE, "hint_tooltip:Smooth shaded corners."), "set_smooth_shaded_corners", "is_smooth_shaded_corners");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "interleave_vertices", PROPERTY_HINT_NONE, "hint_tooltip:Interleave vertices to reduce vertex count."), "set_interleave_vertices", "is_interleave_vertices");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "filter_overlaps", PROPERTY_HINT_NONE), "set_filter_overlaps", "is_filter_overlaps");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "scale_uv_by_length"), "set_scale_uv_by_length", "is_scale_uv_by_length");
+
+	BIND_ENUM_CONSTANT(TESSELLATION_BAKED);
+	BIND_ENUM_CONSTANT(TESSELLATION_DISABLED);
+	BIND_ENUM_CONSTANT(TESSELLATION_ADAPTIVE);
+
+	BIND_ENUM_CONSTANT(PROFILE_FLAT);
+	BIND_ENUM_CONSTANT(PROFILE_CROSS);
+	BIND_ENUM_CONSTANT(PROFILE_TUBE);
+}
+
+void Curve3DMesh::_validate_property(PropertyInfo &p_property) const {
+	if (p_property.name == "interleave_vertices" && profile == PROFILE_TUBE) {
+		p_property.usage = PROPERTY_USAGE_NO_EDITOR;
+	}
+
+	if (p_property.name == "extend_edges" && (!curve.is_valid() || curve->is_closed())) {
+		p_property.usage = PROPERTY_USAGE_NO_EDITOR;
+	}
+
+	if (p_property.name == "segments" && profile == PROFILE_FLAT) {
+		p_property.usage = PROPERTY_USAGE_NO_EDITOR;
+	}
+
+	if (p_property.name == "tile_segment_uv" && profile != PROFILE_CROSS) {
+		p_property.usage = PROPERTY_USAGE_NO_EDITOR;
+	}
+
+	if (p_property.name == "tessellation_tolerance" && tessellation_mode != TESSELLATION_ADAPTIVE) {
+		p_property.usage = PROPERTY_USAGE_NO_EDITOR;
+	}
+
+	if (p_property.name == "scale_uv_by_width" && width_curve.is_null()) {
+		p_property.usage = PROPERTY_USAGE_NO_EDITOR;
+	}
+}
+
+void Curve3DMesh::set_width(const float p_width) {
+	if (width != p_width) {
+		width = p_width;
+		request_update();
+	}
```

Comment:
```suggestion
	if (width == p_width) {
		return;
	}
```suggestion
	if (width != p_width) {
		width = p_width;
		request_update();
	}
```
	width = p_width;
	request_update();
```
Early exit instead, for all these